rerun_py/rerun_sdk/rerun/log/camera.py

```diff
@@ -66,8 +87,50 @@ def log_pinhole(
         Specifies the [`rerun.RecordingStream`][] to use.
         If left unspecified, defaults to the current active data recording, if there is one.
         See also: [`rerun.init`][], [`rerun.set_global_data_recording`][].
+    camera_xyz:
+        Sets the view coordinates for the camera. The default is "RDF", i.e. X=Right, Y=Down, Z=Forward.
+        Other common formats are "RUB" (X=Right, Y=Up, Z=Back) and "FLU" (X=Forward, Y=Left, Z=Up).
+        Equivalent to calling [`rerun.log_view_coordinates(entity, xyz=…)`][rerun.log_view_coordinates].
+        NOTE: setting this to something else than "RDF" (the default) will change the orientation of the camera frustum,
+        and make the pinhole matrix not match up with the coordinate system of the pinhole entity.
+        The pinhole matrix (the `child_from_parent` argument) always project along the Z axis of the camera space,
+        but will be re-oritented to project along another axis if the `camera_xyz` argument is set.
 
     """
 
-    instanced = {"rerun.pinhole": PinholeArray.from_pinhole(Pinhole(child_from_parent, [width, height]))}
+    matrix: npt.ArrayLike
+    if child_from_parent is None:
+        # TODO(emilk): Use a union type for the Pinhole component instead of converting to a matrix here
+        if focal_length_px is None:
+            _send_warning("log_pinhole: either child_from_parent or focal_length_px must be set", 1)
+            focal_length_px = (height * width) ** 0.5  # a reasonable default
+        if principal_point_px is None:
+            principal_point_px = [width / 2, height / 2]
+        if type(focal_length_px) in (int, float):
+            fl_x = focal_length_px
+            fl_y = focal_length_px
+        else:
+            fl_x = focal_length_px[0]  # type: ignore[index]
```

Comment:
We should either check that this is an array-like before indexing, or wrap in a try-catch.

Reply:
my god, run-time tpye-checking dynamically typed languages is a pain in the ass